src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Read.cs

```diff
@@ -376,7 +376,8 @@ private async Task ProcessDataBlockAsync(Stream archiveStream, bool copyData, Ca
                 return null;
             }
 
-            long size = (int)TarHelpers.ParseOctal<uint>(buffer.Slice(FieldLocations.Size, FieldLengths.Size));
+            long size = (long)TarHelpers.ParseOctal<ulong>(buffer.Slice(FieldLocations.Size, FieldLengths.Size));
+            Debug.Assert(size <= TarHelpers.MaxSizeLength, "size exceeded the max value possible with 11 octal digits. Actual size " + size);
```

Comment:
I don't see how this assert could ever happen. The `ParseOctal` method throws if any of the digits is not between 0-7: [TarHelpers.cs#L221](https://github.com/dotnet/runtime/blob/b609e5349c5a5fb4aeee519745491cf60286ef41/src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHelpers.cs#L221)

This means the largest possible number that can be stored in that sliced portion of the header, without throwing, is the expected eleven '7' digits. The assert won't ever catch anything larger.

Reply:
Oh! `FieldLengths.Size` is 12, so if the null char is not respected, the assert would catch if that reserved char were overwritten with an octal number.

Nevermind, all good.